ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/MultiResponseSplitConverter.java

```diff
@@ -0,0 +1,529 @@
+        builder.add(makeRequiredParam(
+                ds3Request,
+                NAMESPACE_FULL_DETAILS,
+                "FullDetails",
+                200,
+                "DetailedTapePartition",
+                null));
+
+        return builder.build();
+    }
+
+    /**
+     * Splits the GetTape request handler into two separate request
+     * handlers. One of the request handlers will have FullDetails removed as an optional
+     * query param with a return type of Tape. The other will have
+     * FullDetails added as a required query param with a return type of TapeApiBean.
+     * @param ds3Request The Get Tape request handler
+     */
+    protected static ImmutableList<Ds3Request> splitGetTape(
+            final Ds3Request ds3Request) {
+        if (!stripPath(ds3Request.getName()).equals(GET_TAPE)) {
+            throw new IllegalArgumentException("The Ds3Request is not the expected "
+                    + GET_TAPE
+                    + ": " + ds3Request.getName());
+        }
+        final ImmutableList.Builder<Ds3Request> builder = ImmutableList.builder();
+        //Add request without Full Details param with a payload of Tape
+        builder.add(removeOptionalParam(
+                ds3Request,
+                "FullDetails",
+                200,
+                "Tape",
+                null));
+
+        //Add request with required Full Details param with a payload of TapeApiBean
+        builder.add(makeRequiredParam(
+                ds3Request,
+                NAMESPACE_FULL_DETAILS,
+                "FullDetails",
+                200,
+                "TapeApiBean",
+                null));
+
+        return builder.build();
+    }
+
+    /**
+     * Modifies a Ds3Request to change a specified optional param into a required
+     * param, name spacing the request name, and modifying a specified response type.
+     * The specified response code will have the specified response type. This is
+     * used when splitting a Ds3Request with multiple response payloads associated
+     * with a given response code.
+     * @param ds3Request A Ds3Request
+     * @param nameSpacing The name spacing used to differentiate this request in the split
+     * @param optionalParamName The name of the optional parameter being modified
+     * @param responseCode The response code to be modified
+     * @param responseType The name of the response type to be associated with the response code
+     * @param responseComponentType The name of the component type to be associated with the
+     *                              the response code. This may be null
+     */
+    protected static Ds3Request makeRequiredParam(
+            final Ds3Request ds3Request,
+            final String nameSpacing,
+            final String optionalParamName,
+            final int responseCode,
+            final String responseType,
+            final String responseComponentType) {
```

Comment:
Is this always `null`?  I don't see this param being used in the above split functions.